lib/ex_unit/lib/ex_unit/on_exit_handler.ex

```diff
@@ -54,6 +54,20 @@ defmodule ExUnit.OnExitHandler do
         {runner_pid, runner_monitor, state || error}
       {:DOWN, ^runner_monitor, :process, ^runner_pid, error} ->
         {nil, nil, state || {{:EXIT, runner_pid}, error, []}}
+    after
+      timeout ->
+        stacktrace =
+          try do
+            Process.info(runner_pid, :current_stacktrace)
+          catch
+            _, _ -> []
+          else
+            {:current_stacktrace, stacktrace} -> stacktrace
+          end
+        Process.exit(runner_pid, :kill)
+        Process.demonitor(runner_monitor, [:flush])
+        exception = ExUnit.TimeoutError.exception(timeout: timeout, type: :on_exit)
+        {nil, nil, state || {:error, exception, stacktrace}}
```

Comment:
The stacktrace should be part of the exception and not used as a fake stacktrace, that is very misleading.